### PR TITLE
Update docs with cookie information

### PR DESCRIPTION
**Pull request by michaelboulton**
https://github.com/taverntesting/tavern/pull/28

### CODE REVIEW

File: documentation.md [hunk 1]
```diff
@@ -569,3 +569,63 @@ response:
...
     returned_uuid: null
 ```
+
+## Persistent cookies
+
+Tavern uses
+[requests](http://docs.python-requests.org/en/master/api/#requests.request)
+under the hood, and uses a persistent `Session` for each test. This means that
+cookies are propagated forward to further stages of a test. Cookies can also be
+required to pass a test. For example, say we have a server that returns a cookie
```
Comment (benhowes) [inline documentation.md]: Mention where the source for this test is?

File: documentation.md [hunk 2]
```diff
@@ -569,3 +569,63 @@ response:
...
+    response:
+      status_code: 200
+      body:
+        name: test-user
+      headers:
+        content-type: application/json
+```
+
+This test ensures that a cookie called `session-cookie` is returned from the
+'login' stage, and this cookie will be sent with all future stages of that test.
```
Comment (benhowes) [inline documentation.md]: Is the term `test` well defined in the document?

File: documentation.md (hunk 1 above)
Comment (michaelboulton) [reply documentation.md]: Which source? This is just an example of a server that would behave like that

File: documentation.md (hunk 2 above)
Comment (michaelboulton) [reply documentation.md]: Yes

File: documentation.md (hunk 1 above)
Comment (benhowes) [reply documentation.md]: `example/cookies/`?

Comment (michaelboulton) [reply documentation.md]: That is just an example implementation of the server in python - it doesn't really relate to the documentation. There aren't implementations for any of the other examples

File: documentation.md (hunk 2 above)
Comment (bitdivision) [reply documentation.md:631]: Are you sure?

**Pull request merged**